connector-commons/http-client/src/main/java/io/camunda/connector/http/client/document/HttpHeaderFilenameResolver.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.http.client.document;
+
+import io.camunda.connector.http.client.client.apache.CustomApacheHttpClient;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.hc.core5.http.HttpHeaders;
+
+public class HttpHeaderFilenameResolver {
+  public static String getFilename(Map<String, Object> headers) {
+    String filename = getFilenameFromContentDispositionHeader(headers);
+    if (!filename.contains(".")) {
+      filename += getFilenameFromContentType(headers);
+    }
+    return filename;
+  }
+
+  private static String getFilenameFromContentDispositionHeader(Map<String, Object> headers) {
+    String contentDispositionHeader =
+        CustomApacheHttpClient.getHeaderIgnoreCase(headers, HttpHeaders.CONTENT_DISPOSITION);
+    if (contentDispositionHeader instanceof String contentDispositionHeaderString) {
+      String filenamePrefix = "filename=";
+      int index = contentDispositionHeaderString.indexOf(filenamePrefix);
+      if (index != -1) {
+        String filename =
+            contentDispositionHeaderString.substring(index + filenamePrefix.length()).trim();
+        if (filename.startsWith("\"") && filename.endsWith("\"")) {
+          filename = filename.substring(1, filename.length() - 1);
+        }
+        return filename;
+      }
+    }
+    return UUID.randomUUID().toString();
+  }
+
+  private static String getFilenameFromContentType(Map<String, Object> headers) {
+    Object ctObj = CustomApacheHttpClient.getHeaderIgnoreCase(headers, HttpHeaders.CONTENT_TYPE);
+    if (ctObj instanceof String ct && ct.contains("/")) {
+      String subtype = ct.substring(ct.indexOf('/') + 1);
+      return "." + subtype;
+    }
+    return "";
+  }
```

Comment:
Is this resolving the filename or the file extension? 

If its for the latter we should rename this method appropriately and the approach might not work out all the time: https://stackoverflow.com/questions/13650372/how-to-determine-appropriate-file-extension-from-mime-type-in-java

Reply:
@ztefanie My concern are mime types like this: [image/svg+xml](https://developer.mozilla.org/en-US/docs/Web/Media/Guides/Formats/Image_types#svg_scalable_vector_graphics) that come with a "plus" sign. I assume we would call a file with this subtype myImage.svg+xm? I think we should only use the first alphanumeric characters without any other signs.

If we take of that I would also be fine skipping the more elaborate solution for now but creating a follow-up issue to fix this holistically (same applies for store document response I assume)